src/domain/position/binProcessor.js

```diff
@@ -88,6 +89,13 @@ const processBins = async (bins, trx) => {
     Enum.Accounts.LedgerEntryType.PRINCIPLE_VALUE
   )
 
+  const latestInitiatingFxTransferInfoByFxCommitRequestId = await BatchPositionModel.getFxTransferInfoList(
+    trx,
+    commitRequestIdList,
+    Enum.Accounts.TransferParticipantRoleType.INITIATING_FSP,
+    Enum.Accounts.LedgerEntryType.PRINCIPLE_VALUE
+  )
+
```

Comment:
I think, the fxTransferParticipant table logic is messed up here https://github.com/mojaloop/central-ledger/blob/0eea335ebe6a04fd0f02ec7d2b8f83ba61b530e8/src/models/fxTransfer/fxTransfer.js#L146
We can't say initiatingFsp is always related to sourceAmount and currency. It depends on the use case.

So, I think its better to get the amount values from participantPositionChange. i.e, findout the fxTransferStateChanges for a commitRequestI, then find out participantPositionChanges for those fxTransferStateChanges filtered by the participant accountID (kafka message key).

Reply:
@vijayg10 done. take another look when you can.